pkg/api/addtask.go

```diff
@@ -0,0 +1,96 @@
+package api
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"time"
+
+	"github.com/c5hark/go_final_project/pkg/db"
+)
+
+type apiError struct {
+	Error string `json:"error"`
+}
+
+type idResponse struct {
+	ID string `json:"id"`
+}
+
+func writeJSON(w http.ResponseWriter, status int, data interface{}) {
+	w.Header().Set("Content-Type", "application/json; charset=utf-8")
+
+	b, err := json.Marshal(data)
+	if err != nil {
+		http.Error(w, fmt.Sprintf(`{"error":"json marshal error: %v"}`, err), http.StatusInternalServerError)
+		return
+	}
+	w.WriteHeader(status)
+	if _, err := w.Write(b); err != nil {
+		return
+	}
+}
+
+// afterNow проверяет, находится ли день t после текущей даты
+func afterNow(now, t time.Time) bool {
+	n := time.Date(now.Year(), now.Month(), now.Day(), 0, 0, 0, 0, time.UTC)
+	return n.After(t)
+}
+
+// checkDate исправляет дату задачи: если задача пустая или прошла, то она обновляется на текущую или следующую дату для повторяющихся задач
+func checkDate(task *db.Task) error {
+	now := time.Now().UTC()
+
+	if task.Date == "" {
+		task.Date = now.Format(dateFormat)
+	}
+
+	t, err := time.Parse(dateFormat, task.Date)
+	if err != nil {
+		return fmt.Errorf("date format error: %v", err)
```

Comment:
При оборачивании нужно использовать `%w` (wrap), а просто при выводе в консоль `%v`
```suggestion
		return fmt.Errorf("date format error: %w", err)
```